src/lib.rs

```diff
@@ -538,6 +541,16 @@ impl Expr {
                 Some(expr) => Ok(ValueType::Option(Box::new(expr.infer_type(scope)?))),
                 None => Ok(ValueType::Option(Box::new(ValueType::Any))),
             },
+            Expr::Append(lhs, rhs) => {
+                let lhs_type = lhs.infer_type(scope)?;
+                let rhs_type = rhs.infer_type(scope)?;
+                if !matches!(lhs_type, ValueType::Seq(_)) {
+                    return Err(anyhow!("Append: lhs is not Seq: {lhs_type:?}"));
+                } else if !matches!(rhs_type, ValueType::Seq(_)) {
+                    return Err(anyhow!("Append: rhs is not Seq: {rhs_type:?}"));
+                }
+                Ok(ValueType::Seq(Box::new(lhs_type)))
+            }
```

Comment:
Does this need to check that the two sequences have the same element type?

Reply:
In looking over this, not only are you 100% correct about the missing unification-check, the return-value is incorrectly wrapped in an extra layer of Seq.